samples/java-application-with-reflection/build.gradle

```
@@ -69,7 +69,10 @@ graalvmNative {
     binaries {
         test {
             verbose = true
-            agent = true
+            agentOptions {
+                enabled = true
+                args.add(providers.systemProperty("agentOptions").forUseAtConfigurationTime())
+            }
```

Now that I've taken a second look at this, it seems a little strange to me to call that block `agentOptions`.

Semantically, it's a container for agent configuration in general.

Also, the `args` are the actual "agent options" -- as in the options supplied to the agent.

What do you think about renaming `agentOptions` to `agent`, `agentConfiguration`, or something similar and renaming `args` to `agentOptions` or simply `options`?

Here's a proposal:

```suggestion
            agent {
                enabled = true
                options.add(providers.systemProperty("agentOptions").forUseAtConfigurationTime())
            }
```

---

We can't use `agent` because there's _already_ a property called `agent` (the old one). I also thought about using it, but we can't really for backwards compatibility. So I moved to `agentOptions` because it's similar to other patterns we find in Gradle: `forkOptions` for example.

---

> We can't use agent because there's already a property called agent (the old one).

---

> We can't use `agent` because there's already a property called `agent` (the old one).

Yes, of course: backwards compatibility.

Sorry I overlooked that.

> So I moved to `agentOptions` because it's similar to other patterns we find in Gradle: `forkOptions` for example.

Since there's a precedent for that, it's OK as-is.

It's just unfortunate that the actual "options" are called "arguments", but in the end the semantics are effectively the same.

---

I could use `options` in the `agentOptions` too, but for some reason I don't like much the redundancy.

---

Understood. Redundancy is bad; naming is difficult. 🤷‍♂️